### PR TITLE
CPDTP-202 Switch from classic autoloader to zeitwerk

### PR DESCRIPTION
**Pull request by blackrat**
### Context

For historical reasons, we have been using the "classic" autoloader. This PR switches it for :zeitwerk

### Changes proposed in this pull request

- change application autoloader to :zeitwerk
- Removed "Concerns" from namespace
- Added in inflector for UTM
- Removed spurious Api:: from the Dummy Api controller

### Guidance to review

Things should behave as before. Only naming changes have been applied.

### Testing

All existing tests pass. Only changes are to namespaces and automatically found names in code where they were previously incorrect, but still being included.

### Review Checks
- [ ] All pages have automated accessibility checks via cypress
- [ ] All pages have visual tests via cypress + percy
- [ ] All `School` queries are correctly scoped - `eligible` when they need to be

### CODE REVIEW

File: spec/controllers/api/api_controller_spec.rb [hunk 1]
```diff
@@ -3,7 +3,7 @@
 require "rails_helper"
 
 class DummyController < Api::ApiController
-  include Api::ApiTokenAuthenticatable
+  include ApiTokenAuthenticatable
 end
```
Comment (kicferk1) [inline spec/controllers/api/api_controller_spec.rb:7]: Why this change?

**Review by kicferk1 — APPROVED** [review]

**Comment by ethanmills** [comment]
Uh oh...

```
/usr/local/bundle/gems/zeitwerk-2.4.2/lib/zeitwerk/loader/callbacks.rb:18:in `on_file_autoloaded': expected file /app/app/decorators/school_decorator.rb to define constant SchoolDecorator, but didn't (Zeitwerk::NameError)
```

File: spec/controllers/api/api_controller_spec.rb (hunk 1 above)
Comment (blackrat) [reply spec/controllers/api/api_controller_spec.rb:7]: It didn't match the actual namespace since it appears in the `concerns` directory which is treated as a root namespace by zeitwerk

**Comment by blackrat** [comment]
> Uh oh...
> 
> ```
> /usr/local/bundle/gems/zeitwerk-2.4.2/lib/zeitwerk/loader/callbacks.rb:18:in `on_file_autoloaded': expected file /app/app/decorators/school_decorator.rb to define constant SchoolDecorator, but didn't (Zeitwerk::NameError)
> ```

Looks like Decorators is also a root namespace. I'll change to drop the module from it.

**Comment by ethanmills** [comment]
"GET /nominations/choose-school" - (127.0.0.1)): #<ActionView::Template::Error: uninitialized constant NominationRequestForm::Decorators>
